config-gen/README.md

```diff
@@ -0,0 +1,81 @@
+# Config-gen
+
+`kubebuilder alpha config-gen` is a subcommand that generates configuration for kubebuilder projects as a configuration function.
+
+Supports:
+
+- Generating CRDs and RBAC from code
+- Generating webhook certificates for development
+- Selectively enabling / disabling components such as prometheus and webhooks
+  - See [types.go](apis/v1alpha1/types.go) for a list of components
+
+## Usage
+
+`config-gen` may be run as a standalone command or from kustomize as a transformer plugin.
+
+### Standalone command
+
+config-gen may be run as a standalone program on the commandline.
+
+See [examples/standalone](examples/standalone/README.md)
+
+### From kustomize
+
+config-gen may be run as a Kustomize plugin using kustomize.
+
+See [examples/kustomize](examples/kustomize/README.md)
+
+### Extending `config-gen`
+
+config-gen may be extended by composing additional functions on top of it.
+
+See examples of layering additional functions on:
+
+- [examples/basicextension](examples/basicextension/README.md)
+- [examples/advancedextension](examples/advancedextension/README.md)
+
+## `KubebuilderProject`
+
+See [types.go](apis/v1alpha1/types.go) for KubebuilderProject schema.
+
+Following is an example of a `kubebuilderproject.yaml` used to generate configuration.
+
+```yaml
+apiVersion: kubebuilder.sigs.k8s.io
+kind: KubebuilderProject
+metadata:
+  name: simple
+spec:
+  # container image containing the controller-manager
+  image: example/simple:latest
+
+  # kubebuilder project root directory relative path from kubebuilderproject.yaml dir
+  # (defaults to current directory)
+  projectDirectory: ../project/...
+
+  # components to enable / disable generating config for
```

Comment:
For the config that's actually commented should we still keep it as it is on the example?

Reply:
I'm open to anything

Reply:
On 2nd thought, since I believe this is auto generated, then let it be as it is then

Reply:
IMO, the problem is that has the potential to grow a lot.  
So, if we scaffold by default all options with the default values it might starts to not be great.
I think @pwittrock is trying to follow up the pattern where when the values are not informed it assumes the default behaviour.

Reply:
This we written by hand.  It is mostly to document the fields that exist.  Only name and image are required, the rest are defaulted.